task6/script.js

```diff
@@ -0,0 +1,507 @@
+        return {};
+    } else{
+        return result;
+    }
+}
+
+function validateAd(ad){
+    if (typeof(ad.id) == 'string' &&
+    typeof(ad.description) == 'string' && ad.description.length < 200  &&
+    ad.createdAt instanceof Date &&  
+    typeof(ad.link) == 'string' &&
+    typeof(ad.vendor) == 'string' && ad.vendor.length != 0 &&
+    ad.hashTags instanceof Array &&
+    typeof(ad.discount) == 'string' &&  ad.discount.length != 0 &&
+    ad.validUntil instanceof Date){
+        return true;
+    }else{
+        return false;
+    }
+}
+
+function addAd(ad){
+    for (let i =0; i<adList.length;++i){
+        if(adList[i].id === ad.id){
+            return false;
+        }
+    }
+    if(validateAd(ad)){
+        adList.push(ad);
+        return true;
+    }else{
+        return false;
+    }
+}
+
+function removeAd(id){
+    for (let i = 0;i<adList.length;++i){
+        if(adList[i].id === id){
+            adList.splice(i,1);
+            return true;
+        }
+    }
+    return false;
+}
+
+function editAd(id,changes){
+    let keys = Object.keys(changes);
+    if(keys.includes('id') || keys.includes('author') || keys.includes('createdAt')){
+        return false;
+    }
+    let ind = adList.findIndex(currentValue => currentValue.id == id);
+    if (ind === -1) {
+        return false;
+    }
+    newAd = Object.assign(adList[ind]);
+    for(let i=0;i<=adList.length;++i){
+        newAd[keys[i]] = changes[keys[i]];
+    }
+    if(validateAd(newAd)){
+        adList[ind] = newAd;
+        return true;
+    }else{
+        return false;
+    }
+
+}
+
+function filterByVendors(ads, vendors,skip, top){
+    return ads.filter(сurrentValue => vendors.includes(сurrentValue.vendor)).slice(skip,top);
+}
+
+function filterByTags (ads, hashTags, skip, top){
+    return ads.filter (currentValue => {
+        for(let j = 0;j<currentValue.hashTags.length;++j){
+            if(hashTags.includes(currentValue.hashTags[j])){
+                return true;
+            }
+        }
+        return false;
+    }).slice(skip,top);
+}
+
+function getAds(skip = 0, top = 10, filterConfig = {}){
+    let adsToShow = [];
+    let keys = Object.keys(filterConfig);
+
+    if(keys.length == 0){
+        adsToShow = adList.slice(skip,top);
+    }else if (keys.length == 1){
+        if(keys[0] == 'hashTags'){
+            adsToShow = filterByTags(adList, filterConfig[keys[0]],skip, top);
+        } else if(keys[0] == 'vendors'){
+            adsToShow = filterByVendors(adList, filterConfig[keys[0]],skip, top);
+        }
+    }else if(keys.length == 2){
+        if(keys[0] == 'hashTags'){
+            adsToShow = filterByTags(adList, filterConfig[keys[0]],skip, top);
+            adsToShow = filterByVendors(adsToShow,filterConfig[keys[1]],0,top);
+        }else if(keys[0] == 'vendors'){
+            adsToShow = filterByVendors(adList, filterConfig[keys[0]],skip, top);
+            adsToShow = filterByTags(adsToShow,filterConfig[keys[1]],0,top);
+        }
```

Comment:
There are two keys in `filterConfig` object, but you still check only first, that's incorrect.